source/world_builder/kd_tree.cc

```diff
@@ -36,7 +36,7 @@ namespace WorldBuilder
                         const bool y_axis)
     {
       const size_t mid=(left+right)>>1;
-      std::nth_element(nodes.begin()+left,nodes.begin()+mid,nodes.begin()+right+1,
+      std::nth_element(nodes.begin()+static_cast<long int>(left),nodes.begin()+static_cast<long int>(mid),nodes.begin()+static_cast<long int>(right)+1,
```

Comment:
What is long int? Is this the same as long long?

Reply:
no, a `long int` is the same as a `long` and a `long long int` is the same as a `long long` (I also had to look it up here: https://en.cppreference.com/w/cpp/language/types). This is apparently what nodes.begin() returns (it is a difference type).